### PR TITLE
BE - PATCH /api/users/:user_id/help-offers

### PR DESCRIPTION
**Pull request by Glawall**
Added rest of files

### CODE REVIEW

File: backend/app/repositories/helpOffers/update.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import db from "../../db/connection";
+
+import { HelpOffer } from "../../db/seeds/data/types/data.types";
+
+export const update = async (user_id: string, helpOfferBody:any): Promise<HelpOffer> => {
```
Comment (Mpanasetckiy) [inline backend/app/repositories/helpOffers/update.ts:5]: user_id we re receiving has to be a number, at least it should be converted in the controller

File: backend/app/repositories/helpOffers/update.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    const {help_request_id, status} = helpOfferBody
+
+    const values = [status, user_id, help_request_id]
+
+    if(values.length ===0){
+        const {rows} = await db.query(
+           `SELECT helper_id, help_request_id, status, users.first_name, users.last_name, users.additional_contacts, users.post_code FROM help_offers LEFT JOIN users on users.id = help_offers.helper_id WHERE helper_id =$1, help_request_id=$2`, [user_id, help_request_id]
+        )
+        return rows[0]
```
Comment (Mpanasetckiy) [inline backend/app/repositories/helpOffers/update.ts:14]: I don't get the point of returning the user? it's part of validation logic which we can implement later in the service

File: backend/app/repositories/helpOffers/update.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+console.log(values)
+
+    const query = `UPDATE help_offers SET status =$1 WHERE helper_id = $2 AND help_request_id=$3 RETURNING helper_id, help_request_id, status`
+
+    const {rows} = await db.query(query, values)
+    console.log(rows)
```
Comment (Mpanasetckiy) [inline backend/app/repositories/helpOffers/update.ts:21]: Could you please make sure no con logs left in the main, it take time to locate them later and remove causing possible conflict if I touch you files :)

File: backend/app/services/helpOffers/update.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import * as helpOffersRepo from "../../repositories/helpOffers/update"
+
+export const update = async(user_id:string, updatedHelpOffer:any) => {
+    const helpOffer = await helpOffersRepo.update(user_id, updatedHelpOffer)
```
Comment (Mpanasetckiy) [inline backend/app/services/helpOffers/update.ts:4]: we can validate here like basic check whether u helpOffer is not undefined

**Review by Mpanasetckiy — APPROVED** [review]
Well done, time to move to the FE

File: backend/app/repositories/helpOffers/update.ts (hunk 2 above)
Comment (Glawall) [reply backend/app/repositories/helpOffers/update.ts:14]: Don't we want to see all that re the UX/Ui design?

File: backend/app/repositories/helpOffers/update.ts (hunk 3 above)
Comment (Glawall) [reply backend/app/repositories/helpOffers/update.ts:21]: Forgot about this one done

File: backend/app/services/helpOffers/update.ts (hunk 4 above)
Comment (Glawall) [reply backend/app/services/helpOffers/update.ts:4]: Yeye I haven't really done any of that, havedone

File: backend/app/repositories/helpOffers/update.ts (hunk 2 above)
Comment (Mpanasetckiy) [reply backend/app/repositories/helpOffers/update.ts:14]: Ahh yeah,  we'll have a look at it later

**Pull request merged**